### ISSUE TITLE
Install failure

### ISSUE THREAD

**Issue by vyktorescu** [williamyang1991/Rerender_A_Video#15]
This looks amazing, looking forward to trying it!
I'm getting a few issues while installing:

**1. No submodules**

When I run `python install.py`, it tells me that it failed to install Ebsynth, but that's because the `/deps/ebsynth` folder is empty. 

I tried running `git clone` on that branch, but that doesn't work—I had to extract the ZIP file into the folder and run `install.py` again.  

Same for Controlnet and gmflow—these folders were empty. I wonder if there's a way to automatically populate the folders during the install, or if it's just me who's having trouble. 

2. **Triton missing**

`C:\Users\sanss\ReRender\Rerender_A_Video>python webUI.py
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'`

3. **Local URL doesn't work**

Despite it telling me to go to http://0.0.0.0:7861, I get ERR_ADDRESS_INVALID in my browser if I try.

**Comment by williamyang1991** [comment]
@SingleZombie, can you help solve this issue?

**Comment by Meatfucker** [comment]
This is due to the initial recursive git clone command not working.

**Comment by SoftologyPro** [comment]
For the 0.0.0.0 URL, use 127.0.0.1 under Windows.

The initial recursive git clone fails.  This is under Windows.

```
Cloning into 'Rerender_A_Video'...
remote: Enumerating objects: 389, done.
remote: Counting objects: 100% (225/225), done.
remote: Compressing objects: 100% (106/106), done.
Receiving objects: 100% (389/389), 9.00 MiB | 5.89 MiB/s, done.sed 164Receiving objects:  92% (358/389), 8.91 MiB | 5.88 MiB/s

Resolving deltas: 100% (222/222), done.
Submodule 'deps/ControlNet' (git@github.com:lllyasviel/ControlNet.git) registered for path 'deps/ControlNet'
Submodule 'deps/ebsynth' (git@github.com:SingleZombie/ebsynth.git) registered for path 'deps/ebsynth'
Submodule 'deps/gmflow' (git@github.com:haofeixu/gmflow.git) registered for path 'deps/gmflow'
Cloning into 'D:/Tests/Rerender_A_Video/deps/ControlNet'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:lllyasviel/ControlNet.git' into submodule path 'D:/Tests/Rerender_A_Video/deps/ControlNet' failed
Failed to clone 'deps/ControlNet'. Retry scheduled
Cloning into 'D:/Tests/Rerender_A_Video/deps/ebsynth'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:SingleZombie/ebsynth.git' into submodule path 'D:/Tests/Rerender_A_Video/deps/ebsynth' failed
Failed to clone 'deps/ebsynth'. Retry scheduled
Cloning into 'D:/Tests/Rerender_A_Video/deps/gmflow'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:haofeixu/gmflow.git' into submodule path 'D:/Tests/Rerender_A_Video/deps/gmflow' failed
Failed to clone 'deps/gmflow'. Retry scheduled
Cloning into 'D:/Tests/Rerender_A_Video/deps/ControlNet'...
Host key verification failed.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:lllyasviel/ControlNet.git' into submodule path 'D:/Tests/Rerender_A_Video/deps/ControlNet' failed
Failed to clone 'deps/ControlNet' a second time, aborting
```

**Comment by SoftologyPro** [comment]
1. I had to cd into the deps folder and git clone the 3 required repositiories.  Then install.py worked.
2. Ignore triton errors under Windows as triton does not have Windows support yet.
3. For Windows use 127.0.0.1 and not 0.0.0.0.

Even with those fixes I am getting other errors when running the examples.  I assume the authors of this do not use Windows.

**Comment by SoftologyPro** [comment]
Running the first example "Run 1st Key Frame" gives this error
```
  File "D:\Tests\Rerender_A_Video\voc_rerender\lib\site-packages\safetensors\torch.py", line 98, in load_file
    with safe_open(filename, framework="pt", device=device) as f:
FileNotFoundError: The system cannot find the file specified. (os error 2)
```

Running the second example "Run 1st Key Frame" does work for the first frame, but when I click "Run All" it does seem to process every frame, but then errors out with
```
  File "D:\Tests\Rerender_A_Video\voc_rerender\lib\site-packages\gradio\components\base.py", line 190, in hash_file
    with open(file_path, "rb") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'result\\pexels-antoni-shkraba-8048492-540x960-25fps\\blend.mp4'
```
The key.mp4 movie exsits, so it seems like an error with the blend step?

Runing the 3rd example "Run 1st Key Frame" gives the same error as the 1st example
```
  File "D:\Tests\Rerender_A_Video\voc_rerender\lib\site-packages\safetensors\torch.py", line 98, in load_file
    with safe_open(filename, framework="pt", device=device) as f:
FileNotFoundError: The system cannot find the file specified. (os error 2)
```

What are we missing under Windows?  I also copied ffmpeg.exe and ffprobe.exe to the same directory as the main script but that made no difference.

**Comment by SingleZombie** [comment]
Please add `--recursive` to your `git clone` command to install the submodules. If you forget it, you can enter `git submodule update --init --recursive` to do the same thing.

**Comment by MichalGeyer** [comment]
Hi! 
I'm facing the same issue and when I used `--recursive ` I got:
`Failed to clone 'deps/gmflow'. Retry scheduled
Cloning into '/home/projects/talide/michagey/projects/Rerender_A_Video/deps/ControlNet'...
Permission denied (publickey).
fatal: Could not read from remote repository.`

Do you know how to solve this?

Thanks and congrats for the great work!

**Comment by nathanshipley** [comment]
I fixed this by modifying the .gitmodules file.  Explained here:  https://github.com/williamyang1991/Rerender_A_Video/issues/20#issuecomment-1723705134

**Comment by oliverban** [comment]
> I fixed this by modifying the .gitmodules file. Explained here: [#20 (comment)](https://github.com/williamyang1991/Rerender_A_Video/issues/20#issuecomment-1723705134)

I've used this fix and installed everything like all the error issues say, 
modify the .gitmodules
install newer gradio
upgrade eiops
in deps/ i have controlnet, ebsynth and gmflow and they seem to be installed correctly (have all files in it)

But when starting the webUI.py it seems to start and gives me an adress that when opened shows **ERR_ADDRESS_INVALID**

**Comment by SoftologyPro** [comment]
> But when starting the webUI.py it seems to start and gives me an adress that when opened shows **ERR_ADDRESS_INVALID**
> 

Use 127.0.0.1 rather than 0.0.0.0 for the URL

**Comment by stevens-Ai** [comment]
Triton missing is because triton is only available on linux and can be ignored as it is not an error and doesnt prevent the code working on windows.

you can use http://localhost:7861/ for the webui

**Comment by oliverban** [comment]
> > But when starting the webUI.py it seems to start and gives me an adress that when opened shows **ERR_ADDRESS_INVALID**
> 
> Use 127.0.0.1 rather than 0.0.0.0 for the URL

Yep, figured that out myself. The thing works up until I want to generate the Keyframes, and there it stops, giving me errors on "Index out of Range" or the likes.

**Comment by dakuohaozyj** [comment]
> I fixed this by modifying the .gitmodules file. Explained here: [#20 (comment)](https://github.com/williamyang1991/Rerender_A_Video/issues/20#issuecomment-1723705134)

git submodule sync
git submodule update --init --recursive Follow the steps step by step. The problem still exists.
